adapter/service/fanctrl/routing_test.go

```diff
@@ -0,0 +1,194 @@
+						},
+					},
+				},
+			},
+			{
+				Name:     "fan ctrl set report",
+				TearDown: adapterhelper.TearDownAdapter("../../testdata/adapter/test_adapter"),
+				Setup:    routeService(modeSetController(t, "night")),
+				Nodes: []*cliffSuite.Node{
+					{
+						Name: "Cmd mode set",
+						Command: cliffSuite.NewMessageBuilder().
+							StringMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:fan_ctrl/ad:2", "cmd.mode.set", "fan_ctrl", "night").
+							Build(),
+						Expectations: []*cliffSuite.Expectation{
+							cliffSuite.ExpectString("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:fan_ctrl/ad:2", "evt.mode.report", "fan_ctrl", "night"),
+						},
+					},
+					{
+						Name: "Device does not exists",
+						Command: cliffSuite.NewMessageBuilder().
+							StringMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:fan_ctrl/ad:404", "cmd.mode.set", "fan_ctrl", "night").
+							Build(),
+						Expectations: []*cliffSuite.Expectation{
+							cliffSuite.ExpectMessage("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:fan_ctrl/ad:404", "evt.error.report", "fan_ctrl"),
+						},
+					},
+					{
+						Name: "Wrong message type",
+						Command: cliffSuite.NewMessageBuilder().
+							FloatMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:fan_ctrl/ad:2", "cmd.mode.set", "fan_ctrl", 1).
+							Build(),
+						Expectations: []*cliffSuite.Expectation{
+							cliffSuite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:fan_ctrl/ad:2", "fan_ctrl"),
+						},
+					},
+				},
+			},
+			{
+				Name:     "broken set mode in controller",
+				TearDown: adapterhelper.TearDownAdapter("../../testdata/adapter/test_adapter"),
+				Setup:    routeService(modeSetBrokenController(t)),
+				Nodes: []*cliffSuite.Node{
+					{
+						Name: "Cmd mode set with error",
+						Command: cliffSuite.NewMessageBuilder().
+							StringMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:fan_ctrl/ad:2", "cmd.mode.set", "fan_ctrl", "night").
+							Build(),
+						Expectations: []*cliffSuite.Expectation{
+							cliffSuite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:fan_ctrl/ad:2", "fan_ctrl"),
+						},
+					},
+				},
+			},
+			{
+				Name:     "broken get mode in controller",
+				TearDown: adapterhelper.TearDownAdapter("../../testdata/adapter/test_adapter"),
+				Setup:    routeService(modeGetBrokenController(t)),
+				Nodes: []*cliffSuite.Node{
+					{
+						Name: "Cmd mode get_report with error",
+						Command: cliffSuite.NewMessageBuilder().
+							StringMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:fan_ctrl/ad:2", "cmd.mode.get_report", "fan_ctrl", "night").
+							Build(),
+						Expectations: []*cliffSuite.Expectation{
+							cliffSuite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:fan_ctrl/ad:2", "fan_ctrl"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	s.Run(t)
+}
+
+func modeSetBrokenController(t *testing.T) *mockedfanctrl.Controller {
```

Comment:
Those helper functions can be moved to 'mocks' to follow already established convention. See https://github.com/futurehomeno/cliffhanger/blob/main/test/mocks/adapter/service/parameters/controller_helper.go as an example